backend/app/views/secondary_donations/_secondary_donation.json.jbuilder

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+json.id secondary_donation.id
+json.amount secondary_donation.amount
+
+if secondary_donation.coupon
+  json.coupon do
+    json.partial! 'coupons/coupon', coupon: secondary_donation.coupon
+  end
+end
```

Comment:
```suggestion
if secondary_donation.coupon
  json.coupon do
    json.partial! 'coupons/coupon', coupon: secondary_donation.coupon
  end
else
  json.coupon.nil
end
```

We should set the JSON value to be `null` if this donation is not attached to a coupon, then let the defined API type be `Nullable<Coupon>`, instead of letting the JSON key be missing.